crates/core/src/config.rs

```diff
@@ -192,42 +219,44 @@ impl ConfigArgs {
         }
 
         let mode = self.mode.unwrap_or(OperationMode::Network);
+        let config_paths = self.config_paths.build(self.id.as_deref())?;
 
         let transport_key = self
             .transport_keypair
-            .map(|path_to_key| {
-                let mut key_file = File::open(&path_to_key).map_err(|e| {
-                    std::io::Error::new(
-                        e.kind(),
-                        format!("Failed to open key file {}: {e}", path_to_key.display()),
-                    )
-                })?;
-                let mut buf = String::new();
-                key_file.read_to_string(&mut buf).map_err(|e| {
-                    std::io::Error::new(
-                        e.kind(),
-                        format!("Failed to read key file {}: {e}", path_to_key.display()),
-                    )
-                })?;
-
-                let pk = rsa::RsaPrivateKey::from_pkcs1_pem(&buf).map_err(|e| {
-                    std::io::Error::new(
-                        std::io::ErrorKind::InvalidData,
-                        format!("Failed to read key file {}: {e}", path_to_key.display()),
-                    )
-                })?;
-
-                Ok::<_, std::io::Error>(TransportKeypair::from_private_key(pk))
-            })
+            .map(Self::read_transport_keypair)
             .transpose()?;
+        let (transport_keypair_path, transport_keypair) =
+            if let Some((transport_key_path, transport_key)) = transport_key {
+                (transport_key_path, transport_key)
+            } else {
+                let transport_key = TransportKeypair::new();
+                let transport_key_path =
+                    config_paths.secrets_dir(mode).join("transport_keypair.pem");
```

Comment:
aren't we allowing passing the path to the transport key by parameter too? in that case if specified, we should be using that ones no?

Reply:
We support passing transport key path as a parameter in `ConfigArgs`, the priority is cli > config file > default.